text-motions.c

```diff
@@ -506,33 +506,40 @@ size_t text_bracket_match(Text *txt, size_t pos) {
 }
 
 static size_t match_symbol(Text *txt, size_t pos, char search, int direction) {
-	char c, current;
+	char c, current, prev_c, next_c;
 	int count = 1;
-	bool instring = false;
+	bool instring = false, inchar = false;
 	Iterator it = text_iterator_get(txt, pos);
 	if (!text_iterator_byte_get(&it, &current))
 		return pos;
+	prev_c = next_c = current;
 	if (direction >= 0) { /* forward search */
 		while (text_iterator_byte_next(&it, &c)) {
-			if (c != current && c == '"')
+			if (!inchar && c != current && c == '"' && prev_c != '\\')
 				instring = !instring;
-			if (!instring) {
+			if (!instring && c != current && c == '\'' && prev_c != '\\')
+				inchar = !inchar;
+			if (!instring && !inchar) {
 				if (c == search && --count == 0)
 					return it.pos;
 				else if (c == current)
 					count++;
 			}
+			prev_c = c;
 		}
 	} else { /* backwards */
 		while (text_iterator_byte_prev(&it, &c)) {
-			if (c != current && c == '"')
+			if (!inchar && c != current && c == '"' && next_c != '\\')
```

Comment:
One of the changes is to ignore quoted, namely `\"`.  When we iterate backward, isn't` \` the next to `"`?  Based on my test with the minimalist examples, this logic works.  A real example is function `text_object_bracket(Text *txt, size_t pos, char type)` in vis's source code file text-objects.c, where jump operator `%` cannot work.